x-pack/plugins/security_solution/public/resolver/view/panels/node_events_of_type.tsx

```diff
@@ -6,295 +6,221 @@
+import React, { memo, useCallback, Fragment } from 'react';
 import { i18n } from '@kbn/i18n';
 import { EuiSpacer, EuiText, EuiButtonEmpty, EuiHorizontalRule } from '@elastic/eui';
 import { useSelector } from 'react-redux';
 import { FormattedMessage } from 'react-intl';
-import styled from 'styled-components';
 import { StyledPanel } from '../styles';
-import { formatDate, StyledBreadcrumbs, BoldCode, StyledTime } from './panel_content_utilities';
-import * as event from '../../../../common/endpoint/models/event';
-import { ResolverEvent, ResolverNodeStats } from '../../../../common/endpoint/types';
+import { formatDate, BoldCode, StyledTime } from './panel_content_utilities';
+import { Breadcrumbs } from './breadcrumbs';
+import * as eventModel from '../../../../common/endpoint/models/event';
+import { SafeResolverEvent } from '../../../../common/endpoint/types';
 import * as selectors from '../../store/selectors';
-import { useResolverDispatch } from '../use_resolver_dispatch';
-import { RelatedEventLimitWarning } from '../limit_warnings';
 import { ResolverState } from '../../types';
-import { useNavigateOrReplace } from '../use_navigate_or_replace';
-import { useRelatedEventDetailNavigation } from '../use_related_event_detail_navigation';
 import { PanelLoading } from './panel_loading';
+import { DescriptiveName } from './descriptive_name';
+import { useLinkProps } from '../use_link_props';
 
 /**
- * This view presents a list of related events of a given type for a given process.
- * It will appear like:
- *
- * |                                                        |
- * | :----------------------------------------------------- |
- * | **registry deletion** @ *3:32PM..* *HKLM/software...*  |
- * | **file creation** @ *3:34PM..* *C:/directory/file.exe* |
+ * Render a list of events that are related to `nodeID` and that have a category of `eventType`.
  */
-
-interface MatchingEventEntry {
-  formattedDate: string;
-  eventType: string;
-  eventCategory: string;
-  name: { subject: string; descriptor?: string };
-  setQueryParams: () => void;
-}
-
-const StyledRelatedLimitWarning = styled(RelatedEventLimitWarning)`
-  flex-flow: row wrap;
-  display: block;
-  align-items: baseline;
-  margin-top: 1em;
-
-  & .euiCallOutHeader {
-    display: inline;
-    margin-right: 0.25em;
-  }
-
-  & .euiText {
-    display: inline;
-  }
-
-  & .euiText p {
-    display: inline;
+export const NodeEventsOfType = memo(
+  ({ nodeID, eventType }: { nodeID: string; eventType: string }) => {
+    const processEvent = useSelector((state: ResolverState) =>
+      selectors.processEventForID(state)(nodeID)
+    );
+    const eventCount = useSelector(
+      (state: ResolverState) => selectors.relatedEventsStats(state)(nodeID)?.events.total
+    );
+    const eventsInCategoryCount = useSelector(
+      (state: ResolverState) =>
+        selectors.relatedEventsStats(state)(nodeID)?.events.byCategory[eventType]
+    );
+    const events = useSelector(
```

Comment:
This technique is something to keep in mind. `useSelector` won't re-run if state and the callback haven't changed. By using `useCallback` you can fully memoize the call.
This won't re-run unless one of these has changed: state, `nodeID`, or `eventType`